.circleci/config.yml

```diff
@@ -22,6 +22,9 @@ commands:
             if [[ $CIRCLE_BRANCH == "master" ]] ; then
               echo "export TINYBIRD_PYTEST_ARGS='--report-to-tinybird '" >> $BASH_ENV
             fi
+            if [[ -n "$CI_PULL_REQUEST" ]] ; then
+              echo "export TESTSELECTION_PYTEST_ARGS='--path-filter=target/testselection/test-selection.txt '" >> $BASH_ENV
+            fi
```

Comment:
We could also create a PR label as a marker that would enforce a full run in any case. This might be a bit confusing though in terms of test execution since the label itself won't trigger the run and you'd need to set it before creating the PR for it to apply on the first run already.

Reply:
Yeah, I guess that could be a bit confusing... Maybe start with a param first?
This seems to be somewhat straight forward: https://circleci.com/docs/reusing-config/#parameter-syntax

Reply:
![image](https://github.com/localstack/localstack/assets/620817/b1d3015f-c3e9-448b-b153-73914fa76014)

The pipeline could then be triggered with this config manually and if set to false the whole test suite is executed. Is that how you envisioned this working?

Reply:
Yeah, exactly! 🚀

Reply:
Tested this now also with the PR at https://github.com/localstack/localstack/pull/10419

https://app.circleci.com/pipelines/github/localstack/localstack?branch=test-selection-lambda

Last 4 runs (1. is the latest, 4. the earliest)

1. **manual** run with `skip_test_selection=true` (=> full run)
2. **manual** run with _no parameters_ set. This run is actually quite interesting because I was curious if a manual run would still have the pull request information available or not. It seems this is the case and therefore the same selection as in 3. applies (=> test selection)
3. run with a change to only the **lambda** service (=> test selection)
4. run with a change to the **IAM** service which isn't in the opt-in list (=> full run)